server/src/test/java/org/apache/cassandra/sidecar/routes/TableStatsHandlerTest.java

```diff
@@ -0,0 +1,204 @@
+
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.TableMetadata;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.google.inject.util.Modules;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.cassandra.sidecar.TestModule;
+import org.apache.cassandra.sidecar.cluster.CassandraAdapterDelegate;
+import org.apache.cassandra.sidecar.cluster.InstancesMetadata;
+import org.apache.cassandra.sidecar.cluster.instance.InstanceMetadata;
+import org.apache.cassandra.sidecar.common.response.TableStatsResponse;
+import org.apache.cassandra.sidecar.common.server.MetricsOperations;
+import org.apache.cassandra.sidecar.server.MainModule;
+import org.apache.cassandra.sidecar.server.Server;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.NOT_FOUND;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the {@link TableStatsHandler} class
+ */
+@ExtendWith(VertxExtension.class)
+public class TableStatsHandlerTest
+{
+
+    private static final int EXPECTED_SSTABLES = 10;
+    private static final long EXPECTED_SIZE = 1024;
+    private static final long EXPECTED_TOTAL_SIZE = 2048;
+    private static final long EXPECTED_SNAPSHOT_SIZE = 100;
+    private static final String KEYSPACE = "testkeyspace";
+    private static final String TABLE = "testtable";
+    static final Logger LOGGER = LoggerFactory.getLogger(TableStatsHandlerTest.class);
+    Vertx vertx;
+    Server server;
+
+    @BeforeEach
+    void before() throws InterruptedException
+    {
+        Module testOverride = Modules.override(new TestModule())
+                                      .with(new TableStatsTestModule());
+        Injector injector = Guice.createInjector(Modules.override(new MainModule())
+                                                        .with(testOverride));
+        server = injector.getInstance(Server.class);
+        vertx = injector.getInstance(Vertx.class);
+        VertxTestContext context = new VertxTestContext();
+        server.start()
+              .onSuccess(s -> context.completeNow())
+              .onFailure(context::failNow);
+        context.awaitCompletion(5, TimeUnit.SECONDS);
+    }
+
+    @AfterEach
+    void after() throws InterruptedException
+    {
+        CountDownLatch closeLatch = new CountDownLatch(1);
+        server.close().onSuccess(res -> closeLatch.countDown());
```

Comment:
Can you use `TestResourceReaper` to tear down resources? 

The class will be available once you rebase. 

```suggestion
        TestResourceReaper.create()
                          .with(server)
                          .with(vertx)
                          .close()
                          .onSuccess(res -> closeLatch.countDown());
```

Reply:
Does this still apply if the tests were moved into the `integration-tests` module to leverage the `SharedCluster` semantics?

Reply:
If it is a integration test, it is less important to close the resources as they are forked per test class. But it is still a good practice to release all resources. 

[TableStatsHandlerTest.java](https://github.com/apache/cassandra-sidecar/pull/197/files#diff-2f6b0450e2df147d6c7e8f8c221f8e61ecd63f22b48a2a7ad1a5adf5b0393630) is not an integration test though.